tools/benchmark/src/MochaCustomOutputReporter.ts

```diff
@@ -0,0 +1,190 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+// This file is a reporter used with node, so depending on node is fine.
+/* eslint-disable import/no-nodejs-modules */
+/* eslint-disable unicorn/prefer-module */
+
+import * as fs from "node:fs";
+import * as path from "node:path";
+
+import chalk from "chalk";
+import Table from "easy-table";
+import { Runner, Suite, Test } from "mocha";
+
+import { isChildProcess } from "./Configuration";
+import { pad, getName } from "./ReporterUtilities";
+// eslint-disable-next-line import/no-internal-modules
+import type { CustomBenchmarkResults } from "./mocha/customOutputRunner";
+// eslint-disable-next-line import/no-internal-modules
+import { getSuiteName } from "./mocha/mochaReporterUtilities";
+
+/**
+ * Custom mocha reporter for benchmarks that provide their own measurements.
+ *
+ * @remarks
+ * It can be used by passing the JavaScript version of this file to mocha's --reporter option.
+ *
+ * This reporter takes output from mocha events and prints a user-friendly version of the results, in addition
+ * to writing them to a file.
+ * The path of the output file can be controlled with --reporterOptions reportDir=<path>.
```

Comment:
Nit (to highlight code syntax):
```suggestion
 * The path of the output file can be controlled with `--reporterOptions reportDir=<path>`.
```